### PR TITLE
MKL_INSTALL_PATH should not be ignore when given

### PR DESCRIPTION
**Pull request by Earthson**
I'm  trying to build tensorflow from source, just found configure will clear my MKL_INSTALL_PATH

### CODE REVIEW

**Comment by Earthson** [comment]
I signed it!

**Comment by Earthson** [comment]
I signed it!

**Review by jhseu — CHANGES_REQUESTED** [review]
This patch does the wrong thing in the ./configure script

Are you sure this is clearing your MKL_INSTALL_PATH? It's not modifying environment variables outside of the ./configure script because it's not calling export.

**Comment by Earthson** [comment]
@jhseu I means this script overwrite the MKL_INSTALL_PATH out side configure, which is expected to working inside.

I'm trying to build a auto compile script like:

```bash
... \
MKL_INSTALL_PATH=$DEST \
... \
./configure
```

**Comment by Earthson** [comment]
@jhseu You misunderstood my intention. I'm trying to make the MKL_INSTALL_PATH configurable outside configure file.

File: configure [hunk 1]
```diff
@@ -226,9 +226,9 @@ if [ "$TF_NEED_MKL" == "1" ]; then # TF_NEED_MKL
   else
     default_mkl_path=/opt/intel/mklml
     fromuser=""
-    read -p "Please specify the location where MKL is installed. [Default is $default_mkl_path]: " MKL_INSTALL_PATH
-    fromuser="1"
     if [ -z "$MKL_INSTALL_PATH" ]; then
+      read -p "Please specify the location where MKL is installed. [Default is $default_mkl_path]: " MKL_INSTALL_PATH
+      fromuser="1"
       MKL_INSTALL_PATH=$default_mkl_path
```
Comment (jhseu) [inline configure]: This overwrites the `read` above.

**Comment by Earthson** [comment]
@jhseu Thank you for the review. That's my mistake.
I think it works now :)

**Comment by Earthson** [comment]
@jhseu Is `$fromuser` set right?  I'm a little confused about it.

**Review by jhseu — APPROVED** [review]
The fromuser part is unnecessary, but since it exists in the old code, no need to fix it here.

**Comment by fayeshine** [comment]
@jhseu Any status on this?

**Comment by jhseu** [comment]
Jenkins, test this please

**Comment by Earthson** [comment]
```bash
//tensorflow/python:rmsprop_test                                         PASSED in 30.6s
//tensorflow/python:saver_test                                           PASSED in 33.5s
//tensorflow/python:session_benchmark                                    PASSED in 20.0s
//tensorflow/python:session_manager_test                                 PASSED in 91.6s
//tensorflow/python:slot_creator_test                                    PASSED in 24.9s
//tensorflow/python:special_math_ops_test                                PASSED in 25.5s
//tensorflow/python:split_benchmark                                      PASSED in 21.3s
//tensorflow/python:tensorboard_logging_test                             PASSED in 19.3s
//tensorflow/python:timeline_test                                        PASSED in 26.6s
//tensorflow/python:training_ops_test                                    PASSED in 26.8s
//tensorflow/python:transpose_benchmark                                  PASSED in 18.5s
//tensorflow/python/kernel_tests:stage_op_test                          TIMEOUT in 1 out of 2 in 329.1s
  Stats over 2 runs: max = 329.1s, min = 35.4s, avg = 182.2s, dev = 146.8s
  /var/lib/jenkins/workspace/tensorflow-pull-requests-gpu/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/workspace/bazel-out/local_linux-opt/testlogs/tensorflow/python/kernel_tests/stage_op_test/shard_1_of_2/test.log

Executed 311 out of 311 tests: 310 tests pass and 1 fails locally.

Parameterized build ends with FAILURE at: Mon Jun  5 22:01:34 UTC 2017 (Elapsed time: 2508 s)
Build step 'Execute shell' marked build as failure
[Set GitHub commit status (universal)] ERROR on repos [] (sha:2dc13c0) with context:tensorflow-pull-requests-gpu
Unable to get pull request builder trigger!!
Setting status of c6e36aa11089ab7283a8a421016c86d191f1220c to FAILURE with url https://ci.tensorflow.org/job/tensorflow-pull-requests-gpu/5116/ and message: 'FAILURE
 '
Using context: Linux GPU
Finished: FAILURE
```

I have no idea about the failure

**Comment by Earthson** [comment]
Jenkins, test this please

**Comment by Earthson** [comment]
@jhseu

**Comment by Earthson** [comment]
Jenkins, test this please

**Comment by jhseu** [comment]
Jenkins, test this please

**Pull request merged**